enhancements/microshift/microshift-router-configuration.md

```diff
@@ -0,0 +1,466 @@
+access to ports. Since this is a feature that users may require for auditing,
+an alternative is provided.
+
+The `NetworkPolicy` resource fits the needs for blocking traffic to services
+and pods. Policies can also be configured to allow specific IP addresses and
+provide a high degree of customization for network traffic. They use the CNI
+instead of host level settings, providing greater flexibility and being self
+contained within MicroShift.
+
+Users in need of blocking traffic shall use the `NetworkPolicy` resource. A
+section in the documentation will be added to describe how to do this.
+
+### Configure IPs where the router listens
+The following configuration is proposed:
+```yaml
+ingress:
+  expose: # Defaults to IPs in the host. Details below.
+    hostnames:
+    - <hostname>
+    interfaces:
+    - <NIC name>
+    ipAddresses:
+    - <IP address>
+```
+As we have seen in previous sections, the use of `LoadBalancer` makes ovnk
+configure iptables rules to expose the service outside of the cluster.
+Building on [LoadBalancer controller enhancement](loadbalancer-service-support.md),
+the service is exposed using the node IP and configured ports.
+
+All the IPs where the router is exposed need to be part of the `LoadBalancer`
+service status definition. In order to provide more flexibility, the
+configuration options are able to take IP addresses, hostnames and/or NIC
+names.
+
+MicroShift will default to listen to external IPs, the service IP and the
+apiserver IP, as this was the previous behavior and it needs to keep it for
+compatibility. Any advanced configuration, such as multiple interfaces, VLANs,
+etc. will need user's configuration.
+
+The hostnames will be automatically resolved to their IP addresses by
+MicroShift, same as the NIC names.
+
+### Allow specific IP addresses
+This may be seen as a special case of firewalling, but using IP addresses
+instead of ports. Using the same `NetworkPolicy` resources, IPs may be
+specified to allow traffic from them and reject it from any other source.
+
+Please note this is a manual procedure for the user. Documentation will be
+added on how to perform the required actions.
+
+More information in the design details section.
+
+### Workflow Description
+**cluster admin** is a human user responsible for configuring a MicroShift
+cluster.
+
+1. The cluster admin adds specific configuration for the router prior to
+   MicroShift's start.
+2. After MicroShift started, the system will ingest the configuration and setup
+   everything according to it.
+3. The router will enabled/disabled, be exposed on the specified ports, on the
+   specified IPs, allowing only specified IP addresses, all according to the
+   cluster admin provided configuration. This includes the status in the
+   LoadBalancer service type for the router.
+
+### API Extensions
+As described in the proposal, there is an entire new section in the configuration:
+```yaml
+ingress:
```

Comment:
I'm wondering if it would be possible to also present the `Service` that will be created out of `ingress`?

Reply:
It's what caused this request - it would be nice to have some visual mapping unless it's little value due to dynamic hostname->IP conversions